documentation/docs/requirements.md

```diff
@@ -0,0 +1,85 @@
+# Overview
+The _Modu_ tool makes use of _Poetry_ to install the majority of the requirements needed. To run the _Modu_ tool itself, these are the only dependencies needed. 
+
+As part of the project was implementing an existing quality model, we have a couple of other dependencies that will need to be installed to run our re-creation of the quality model that Code Climate makes use of
+
+## Installing poetry and dependencies
+### Python
+Our tool will require a user to have a python version of 3.10 or higher. Please ensure that you are making use of an appropriate python version
+### Python Poetry
+As mentioned before, _Modu_ makes use of _Poetry_ to manage the majority of our dependencies. Please refer to the poetry documentation on how to install it: https://python-poetry.org/docs/
+### Tree-sitter
+_Tree-sitter_ is used for generating abstract syntax tree that can be used for querying specific node types. _Tree-sitter_ requires specifically built grammar types for each coding language analyzed. These grammars can be generated by doing the following: 
+
+Ensure that you are in the **tool** folder. 
+```
+tool/
+    ...
+    documentation/
+    figures/
+    modu/
+    ...
+```
+
+Run the following commands:
+Generation of build and grammars folder:
+```sh
+mkdir build grammars
+```
+Clone the different grammars into grammars/:
+```sh
+cd grammars && git clone https://github.com/tree-sitter/tree-sitter-python && git clone https://github.com/tree-sitter/tree-sitter-c-sharp && git clone https://github.com/tree-sitter/tree-sitter-java && cd -
+```
+Build the grammar file:
+```sh
+poetry run python modu/build_treesitter.py
+```
+
+**Adding new languages**
+
+If you want to expand on the current capabilities of the program by adding a new supported language, you will need to do the following: 
+Addding a new language to the grammars folder (please refer to the _Tree-sitter_ documentation to see supported languages: https://tree-sitter.github.io/tree-sitter/):
+```sh
+cd grammars && git clone https://github.com/tree-sitter/tree-sitter-<some-supported-language> && cd -
+```
+Adding new grammar to build command: 
+```sh
+cd modu
+vim build_treesitter.py
+-------------------------
+Language.build_library(
+    "build/my-languages.so",
+    [
+        "grammars/tree-sitter-python",
+        "grammars/tree-sitter-c-sharp",
+        "grammars/tree-sitter-java",
+        "grammars/<your-new-language>"
+    ],
+)
+--------------------------
+```
+## Other dependencies (Code Climate Implementation)
+### PMD - Copy Paste Detector (CPD)
+Part of implementing code climate was finding the amount of similar and identical code blocks. We managed this by using an external tool called PMD or more specifically Copy Paste Detector (CPD) - https://docs.pmd-code.org/latest/index.html
+
+We made use of the 6.54.0 (January 2023) version, but please do check if there are any newer releases: https://github.com/pmd/pmd/releases
```

Comment:
I think we _should_ encourage the exact same version of CPD to be used that we have used and tested.

Reply:
alrighty